src/SelfServe/SqlX/SqlX.tsx

```diff
@@ -9,91 +10,325 @@ import {
+  });
+  currentValues.set("instances", {
+    value: instances.value,
+    hidden: hideAttributes,
+    disabled: dedicatedGatewayOriginallyEnabled,
+  });
+
+  currentValues.set("skuDetails", {
+    value: { textTKey: getSKUDetails(`${currentValues.get("sku").value}`), type: DescriptionType.Text } as Description,
+    hidden: hideAttributes,
+    disabled: dedicatedGatewayOriginallyEnabled,
+  });
+
+  currentValues.set("costPerHour", { value: costPerHourValue, hidden: hideAttributes });
+  currentValues.set("connectionString", {
+    value: connectionStringValue,
+    hidden: !newValue || !dedicatedGatewayOriginallyEnabled,
+  });
+
+  return currentValues;
 };
 
+const skuDropDownItems: ChoiceItem[] = [
+  { label: "CosmosD4s", key: "Cosmos.D4s" },
+  { label: "CosmosD8s", key: "Cosmos.D8s" },
+  { label: "CosmosD16s", key: "Cosmos.D16s" },
+  { label: "CosmosD32s", key: "Cosmos.D32s" },
+];
+
 const getSkus = async (): Promise<ChoiceItem[]> => {
-  // TODO: get SKUs from getRegionSpecificSkus() RP call and return array of {label:..., key:...}.
-  throw new Error("getSkus not implemented.");
+  return skuDropDownItems;
 };
 
 const getInstancesMin = async (): Promise<number> => {
-  // TODO: get SKUs from getRegionSpecificSkus() RP call and return array of {label:..., key:...}.
-  throw new Error("getInstancesMin not implemented.");
+  return 1;
 };
 
 const getInstancesMax = async (): Promise<number> => {
-  // TODO: get SKUs from getRegionSpecificSkus() RP call and return array of {label:..., key:...}.
-  throw new Error("getInstancesMax not implemented.");
-};
-
-const validate = (currentValues: Map<string, SmartUiInput>): void => {
-  // TODO: add cusom validation logic to be called before Saving the data.
-  throw new Error(`validate not implemented. No. of properties to validate: ${currentValues.size}`);
+  return 5;
 };
 
 @IsDisplayable()
+@RefreshOptions({ retryIntervalInMs: 20000 })
 export default class SqlX extends SelfServeBaseClass {
   public onRefresh = async (): Promise<RefreshResult> => {
-    return refreshDedicatedGatewayProvisioning();
+    return await refreshDedicatedGatewayProvisioning();
   };
 
-  public onSave = async (currentValues: Map<string, SmartUiInput>): Promise<OnSaveResult> => {
-    validate(currentValues);
-    // TODO: add pre processing logic before calling the updateDedicatedGatewayProvisioning() RP call.
-    throw new Error(`onSave not implemented. No. of properties to save: ${currentValues.size}`);
+  public onSave = async (
+    currentValues: Map<string, SmartUiInput>,
+    baselineValues: Map<string, SmartUiInput>
+  ): Promise<OnSaveResult> => {
+    const dedicatedGatewayCurrentlyEnabled = currentValues.get("enableDedicatedGateway")?.value as boolean;
+    const dedicatedGatewayOriginallyEnabled = baselineValues.get("enableDedicatedGateway")?.value as boolean;
+
+    currentValues.set("warningBanner", undefined);
+
+    //TODO : Add try catch for each RP call and return relevant notifications
```

Comment:
Do you want to catch any errors and give a well formed error message via a new error? right now we just show whatever is returned as part of the thrown error.
try {
deleteDedicatedGateway(...)
} catch (e) {
  if (e.message === "...") {
    throw new Error ("well formed error message")
   }
  throw e
}

Reply:
This is okay, because the RP message is what is propagated in the other services, like "exclusive account lock" etc... 

Unless there is a need, I don't think we need to pre-emptively add some message.